### ISSUE TITLE
About the contrastive loss

### ISSUE THREAD

**Issue by JTYSXYYYBFDLJZ** [QinbinLi/MOON#14]
This is an excellent piece of work！
However, when I read the code，Why does multiplying logits with an all-zero matrix cause the contrast loss to always be 0？
            _, pro1, out = net(x)
            _, pro2, _ = global_net(x)

            posi = cos(pro1, pro2)
            for previous_net in previous_nets:
                previous_net.cuda()
                _, pro3, _ = previous_net(x)
                nega = cos(pro1, pro3)
                logits = torch.cat((logits, nega.reshape(-1,1)), dim=1)
                previous_net.to('cpu')
            logits /= temperature
            labels = torch.zeros(x.size(0)).cuda().long()
            loss2 = mu * criterion(logits, labels)###here

**Comment by QinbinLi** [comment]
Hi @JTYSXYYYBFDLJZ ,

Please refer to #8. The contrastive loss can be computed by cross-entropy loss.
